internal/provider/provider.go

```diff
@@ -159,12 +178,34 @@ func configure(version string, p *schema.Provider) func(context.Context, *schema
 			})
 		}
 
+		// Normal Client
 		client := &AWSClient{
 			accountid:         accountID,
 			organizationsconn: organizations.New(sess.Copy()),
 			scconn:            servicecatalog.New(sess.Copy()),
 		}
 
+		/// If we have an assume role ARN then use stscreds, otherwise continue as normal...
+		if _, ok := d.GetOk("assume_role"); ok {
+
+			assumeroleblock := d.Get("assume_role").([]interface{})[0]
+
+			if assumeroleblock != nil {
+
+				arn := assumeroleblock.(map[string]interface{})["role_arn"].(string)
+
+				stsconfig := &aws.Config{Credentials: stscreds.NewCredentials(sess, arn)}
+
+				client = &AWSClient{
+					accountid:         accountID,
+					organizationsconn: organizations.New(sess.Copy(stsconfig)),
+					scconn:            servicecatalog.New(sess.Copy(stsconfig)),
+				}
```

Comment:
Instead of this block you can also use the features of Hashicorp's aws-sdk-go-base by simply setting the `AssumeRoleARN` of the awsbase config. Something like this:

```go
config.AssumeRoleARN = arn 
```

Note that this would need to go above the `GetSessionWithAccountIDAndPartition` call.

Reply:
Had a feeling there must have been an easier way! Thanks